.github/scripts/findUnusedStyles.ts

```diff
@@ -0,0 +1,462 @@
+import * as fs from 'fs';
+import {globSync} from 'glob';
+import * as path from 'path';
+import * as ts from 'typescript';
+
+type StyleDefinition = {
+    key: string;
+    file: string;
+    line: number;
+    column: number;
+};
+
+// Static patterns and constants - created once
+const STYLE_FILE_EXTENSIONS = '**/*.{ts,tsx,js,jsx}';
+const EXCLUDED_STYLE_DIRECTORIES = ['src/styles/utils/', 'src/styles/generators/', 'src/styles/theme/'];
+const VALID_IDENTIFIER_PATTERN = /^[a-zA-Z_$][a-zA-Z0-9_$]*$/;
+const SPREAD_PROPERTY_PATTERN = /\.\.\.\s*([a-zA-Z_$][a-zA-Z0-9_$]*)\s*\.\s*([a-zA-Z_$][a-zA-Z0-9_$]*)/g;
+const SIMPLE_SPREAD_PATTERN = /\.\.\.\s*([a-zA-Z_$][a-zA-Z0-9_$]*)\b(?!\s*\.)/g;
+const STYLE_KEY_SKIP_PATTERNS = ['default', 'exports', 'module', 'require', 'import', 'from', 'theme', 'colors', 'variables', 'CONST', 'Platform', 'StyleSheet'];
+
+// Comment removal patterns
```

Comment:
Hmmm I definitely don't see why you should need this. If memory serves, TS ast nodes have two properties: `getText` and `getFullText`, and `getFullText` includes comments but `getText` does not.